routes/fire_weather.py

```diff
@@ -0,0 +1,753 @@
+        op: postprocessing operation to perform (required)
+            valid operations: 3_day_rolling_average, 5_day_rolling_average, 7_day_rolling_average, summer_fire_danger_rating_days
+            only one operation can be performed at a time
+        format: output format (only supports "csv")
+
+    Args:
+        place_id (str): place identifier, used to fetch polygon and compute zonal statistics
+        start_year (int): optional start year for WCPS query
+        end_year (int): optional end year for WCPS query
+
+    Returns:
+        JSON-like dict of requested daily fire weather data
+
+    Notes:
+        usage examples (HUC10 polygon: 1908030906):
+        - 3 day rolling average for all variables, all years: http://localhost:5000/fire_weather/area/1908030906
+        - 5 day rolling average for select variables, all years: http://localhost:5000/fire_weather/area/1908030906?vars=bui,fwi&op=5_day_rolling_average
+        - summer fire danger rating days for select years: http://localhost:5000/fire_weather/area/1908030906/2000/2030?op=summer_fire_danger_rating_days
+        - summer fire danger rating days for select variables, select years: http://localhost:5000/fire_weather/area/1908030906/2000/2030?vars=bui,fwi&op=summer_fire_danger_rating_days
+    """
+
+    poly_type = validate_var_id(place_id)
+    if type(poly_type) is tuple:
+        return poly_type
+
+    try:
+        polygon = get_poly(place_id, crs=4326)
+    except:
+        return render_template("422/invalid_area.html"), 422
+
+    # validate that the polygon is completely within the geotiff that represents the footprint of the data coverage
+    if check_poly_in_geotiffs(polygon, coverages=[fire_weather_geotiff]) == 404:
+        return render_template("404/no_data.html"), 404
+
+    requested_vars = request.args.get("vars")
+    requested_vars = validate_vars(requested_vars)
+
+    start_year = int(start_year) if start_year is not None else None
+    end_year = int(end_year) if end_year is not None else None
+    var_time_slices = validate_requested_vars_start_and_end_year(
+        requested_vars, start_year, end_year
+    )
+
+    requested_ops = request.args.get("op")
+    requested_ops = validate_postprocessing_operation(requested_ops)
+
+    try:
+        # fetch bbox datasets for requested variables
+        datasets_dict = asyncio.run(
+            fetch_polygon_data_for_all_vars(requested_vars, polygon, var_time_slices)
+        )
+        zonal_results = calculate_zonal_stats(polygon, datasets_dict, requested_vars)
+    except Exception as exc:
+        if hasattr(exc, "status") and exc.status == 404:
+            return render_template("404/no_data.html"), 404
+        return render_template("500/server_error.html"), 500
+
+    n = None
+    if "3_day_rolling_average" in requested_ops:
+        n = 3
+    elif "5_day_rolling_average" in requested_ops:
+        n = 5
+    elif "7_day_rolling_average" in requested_ops:
+        n = 7
```

Comment:
This gets repeated. I'd suggest making this a dict that is constant at the top of this file, or, dunking it in `luts.py`.